md2html.sh

```diff
@@ -17,44 +15,33 @@ while getopts "htl" opt; do
     l)
       convert_link=1
       shift ;;
-    t)
-      add_toc=1
-      toc_arg='--toc-levels 2..6'
-      shift ;;
     *)
       ;;
   esac
 done
 
 for file in "$@"; do
-  md2html_tool=""
   # RHEL 9 in brew cannot use pandoc, hence trying kramown first
-  if type -p kramdown >/dev/null && kramdown -i GFM < /dev/null > /dev/null; then
-    md2html_tool=kramdown
-  elif type -p pandoc >/dev/null; then
-    md2html_tool=pandoc
+  if [ -z "${md2html_tool:-}" ]; then
+    if type -p kramdown >/dev/null && kramdown -i GFM < /dev/null > /dev/null; then
+      md2html_tool=kramdown
+    elif type -p pandoc >/dev/null; then
+      md2html_tool=pandoc
+    fi
   fi
-  # With kramdown, convert directly to HTML
   if [ "$md2html_tool" == kramdown ]; then
-    if [ "$add_toc" -ne 0 ]; then
-      sed -i '1s/^/* toc\n{:toc}\n/' "${file}"
-    fi
+    sed -i '1s/^/* toc\n{:toc}\n/' "${file}"
     # Set locale to UTF-8 because by default it is set to US-ASCII
     # shellcheck disable=SC2086
     LC_ALL=C.UTF-8 $md2html_tool --extension parser-gfm \
       --input GFM \
-      $toc_arg \
+      --toc-levels "2..6" \
       --output html "${file}" > "${file%.md}.html"
-    if [ "$add_toc" -ne 0 ]; then
-      sed -i '1,2d' "${file}"
-    fi
-  # With pandoc, convert to adoc, then to HTML
+    sed -i '1,2d' "${file}"
   elif [ "$md2html_tool" == pandoc ]; then
-    $md2html_tool -f gfm "${file}" -t asciidoc -o "${file%.md}.tmp.adoc"
-    touch -r "${file}" "${file%.md}.tmp.adoc"
-    TZ=UTC asciidoc -o "${file%.md}.html" -a footer-style=none -a toc2 -a source-highlighter=highlight "${file%.md}.tmp.adoc"
-    tr -d '\r' < "${file%.md}.html" > "${file%.md}.tmp.adoc"
-    mv "${file%.md}.tmp.adoc" "${file%.md}.html"
+    $md2html_tool --from gfm "${file}" --to html --output "${file%.md}.html" \
+      --toc --shift-heading-level-by=-1 \
+      --template https://raw.githubusercontent.com/tajmone/pandoc-goodies/master/templates/html5/github/GitHub.html5
```

Comment:
I guess if we ever want to use this downstream or in rpm build, we'll need to package this file

Reply:
Oh yeah, I need to make this conditional so that we do not use this in RPM build for now. My plan was to get rid of it from the rpm build completely because there is not pandoc on RHEL 9+.